service-base/src/main/java/org/eclipse/hono/service/monitoring/ConnectionEventProducer.java

```diff
@@ -14,38 +14,49 @@
  * <p>
  * The interface is intended to be implemented by setups which are interested in receiving technical connection events.
  * Which might not make sense for all protocols adapters. But mostly for connection oriented ones like MQTT or AMQP 1.0.
  * <p>
- * The protocol adapters may call {@link #connected(String, Device)} and {@link #disconnected(String, Device)} as they
- * see fit. The whole process is a "best effort" process and intended to be used for monitoring/debugging.
+ * The protocol adapters may call {@link #connected(String, String, Device, JsonObject)} and
+ * {@link #disconnected(String, String, Device, JsonObject)} as they see fit. The whole process is a "best effort"
+ * process and intended to be used for monitoring/debugging.
  * <p>
  * When a protocol adapter calls into this producer it must provide some kind of connection ID, which might have a
  * different meaning for different protocol adapters. This should be as identifiable as possible but there is no
  * requirement for this to be unique. However for each connection the protocol adapter must call
- * {@link #disconnected(String, Device)} with the same information as it called {@link #connected(String, Device)}.
+ * {@link #disconnected(String, String, Device, JsonObject)} with the same information as it called
+ * {@link #connected(String, String, Device, JsonObject)}.
  */
 public interface ConnectionEventProducer {
 
     /**
      * Produce an event for a new connection.
      * 
      * @param remoteId The ID of the remote endpoint which connected (e.g. a remote address, port, client id, ...).
+     * @param protocolAdapter The name of the protocol adapter sending this event. Must not be {@code null}.
      * @param authenticatedDevice The optional authenticated device associated with the connection. May be {@code null}
      *            if the connection is from an unauthenticated device.
+     * @param data Additional, protocol adapter specific data
+     * @return A future which indicates the result of the event production
      */
-    void connected(String remoteId, Device authenticatedDevice);
+    Future<?> connected(String remoteId, String protocolAdapter, Device authenticatedDevice, JsonObject data);
 
     /**
      * Produce an event for a closed connection.
      * 
      * @param remoteId The ID of the remote endpoint which disconnected. The same ID used in the call to
-     *            {@link #connected(String, Device)}
+     *            {@link #connected(String, String, Device, JsonObject)}
+     * @param protocolAdapter The name of the protocol adapter sending this event. Must not be {@code null}.
      * @param authenticatedDevice The optional authenticated device associated with the connection. May be {@code null}
      *            if the connection is from an unauthenticated device.
+     * @param data Additional, protocol adapter specific data
+     * @return A future which indicates the result of the event production
```

Comment:
There seem to be some required params for which we usually add a `@throws NullPointerException` tag to make clear what to expect ...

Reply:
Added.